### DISCUSSION TITLE
Magisk Canary not available

### DISCUSSION THREAD

**Discussion by supermansaga** [LSPosed/MagiskOnWSALocal#608] [Q&A]
On the pink screen, I selected Canary, meaning supposedly 26103 now, but all I got is still Stable 26100. Thanks. Also, if I use Shamiko 0.7.3 and Displax‘s SafetynetFix [v2.4.0-MOD_1.3] https://github.com/Displax/safetynet-fix/releases, I still can't past safetynetfix? Thx

**Comment by Howard20181** [comment]
1. Delete magisk-canary.zip in the download folder
2. No way to pass SafetyNet on emulator


Best regards,

Howard
________________________________
From: supermansaga ***@***.***>
Sent: Sunday, July 2, 2023 2:28:43 PM
To: LSPosed/MagiskOnWSALocal ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [LSPosed/MagiskOnWSALocal] Magisk Canary not available (Discussion #608)


On the pink screen, I selected Canary, meaning supposedly 26103 now, but all I got is still Stable 26100. Thanks. Also, if I use Shamiko 0.7.3 and Displax‘s SafetynetFix [v2.4.0-MOD_1.3] https://github.com/Displax/safetynet-fix/releases, I still can't past safetynetfix? Thx

**Reply by supermansaga** [reply]
Got it. But 

1. what do you mean "delete magisk-canary.zip in the download folder"? Is it because that's not the true 26103? And we could replace it with the real deal? Trying to understand how to do so as a Ubuntu newbie.

2. I guess given no way to pass SafetyNet, no point in trying to hide Magisk? I broke mine earlier by hiding....

Thanks.

**Comment by Howard20181** [comment]
1. In the download process aria2 should replace the new canary magisk but I don't know why it doesn't. So, remove it manually.
2. You can hide Magisk, but hidden magisk not mean passed SafetyNet.


Best regards,

Howard
________________________________

**Reply by supermansaga** [reply]
1. Do you mean I can remove it manually? Assuming I can, I just don't see that file in the folder called WSA_2305.40000.6.0_x64_Release-Nightly-with-magisk-26.1(26100)-stable-MindTheGapps-13.0-RemovedAmazon, after running through the pink screen. I perform a search for *magisk* on windows 11 after copy pasting the folder over. And found nothing. 

2. I understand now. Yasnac, Integrity checker all show failed. Could passing safety net be something achievable down the road though? 

3. Is it true that the latest WSA is just a Pixel 5? Could Pixel 7 Pro be installed instead?

Thanks.

**Comment by Howard20181** [comment]
1. The download folder should place in the same as the MagiskOnWSA build scripts folder.
2. No way to pass SafetyNet on the emulator.
3. change it to another in the fix props python script.

**Reply by supermansaga** [reply]
1. By "same," I think you meant they are parallel to each other. I do see such folder now and magisk-stable.zip is in there instead of magisk-canary.zip, even after I choose magisk canary debug. But at this point, even if I knew how to replace stable w/ canary, how would it affect what's already in the output folder?

3. Changing Pixel 5 to Pixel 7 Pro is just a cosmetic thing instead of downloading the real P7P's factory image? The only place where I can see what model is to open "files" within the "system" tab of WSA. If it's purely cosmetic, no point in changing it? But I did receive an email from google, congratulating me that I have pixel 5. So it appears it's more than cosmetic. If this is more than cosmetic, how can I actually change it in the script you mentioned, could you please lay out some more detailed steps? I am just a newbie in ubuntu/linux. 

Many thanks

**Reply by Howard20181** [reply]
1. You should use the space key to select
2. just change the props

**Reply by supermansaga** [reply]
1. I think I do understand what you meant by "space key" now. I have got "WSA_2305.40000.6.0_x64_Release-Nightly-with-magisk-a8c4a33e(26103)-canary-MindTheGapps-13.0-RemovedAmazon" instead
2. Exactly how do I change the props? Could you please elaborate a bit more? I just can't find it and know which command to type. Thanks.

**Reply by fromhgbwithluv** [reply]
@supermansaga You can check this pull request from my repos, which is support to change Pixel 5 to any other Pixel
https://github.com/YT-Advanced/WSA-Script/pull/79/files
Or if you are too lazy, you can build by yourself follow this guide:
https://github.com/YT-Advanced/WSA-Script#--want-to-build-your-custom-build-

**Reply by fromhgbwithluv** [reply]
You need to change prop from these path, and also replace all the Build ID with the data from `ro.build.id`:
```javascript
/system/build.prop
```
```javascript
/vendor/build.prop
```
```javascript
/vendor/odm/etc/build.prop
```
```javascript
/vendor/vendor_dlkm/etc/build.prop
```

**Reply by supermansaga** [reply]
@YT-Advanced thanks! But what exactly does this change achieve? What additional functionality does it have after upgrading to pixel 7 pro? Would it allow me to play Super Mario Run downloaded from Google Play Store then? These paths you mentioned...where are they located? Within the folder called "WSA_2305.40000.6.0_x64_Release-Nightly-with-magisk-a8c4a33e\(26103\)-canary-MindTheGapps-13.0-RemovedAmazon/"? Apologies for these stupid questions. Your solution still seems way too complex for me. I wish the original team could have added this selection of pixel devices on the pink screen. Thanks.

**Reply by fromhgbwithluv** [reply]
@supermansaga 
1. Change from Pixel 5 to Pixel 7 may help some developer to test their app (I don't know what they do with it, but somebody have suggested me to add this feature)
2. These path only can be found when mounting VHDX file, which is hard for normal user, so I think you should create your own Custom Build follow this guide:
https://github.com/YT-Advanced/WSA-Script#--want-to-build-your-custom-build-
3. I'm not online with my PC so I can't test Super Mario, you can build, download and test by yourself
